seed/static/seed/js/controllers/insights_property_controller.js

```diff
@@ -290,33 +290,49 @@ angular.module('BE.seed.controller.insights_property', [])
           } else if (_.includes($scope.data.results_by_cycles[$scope.configs.chart_cycle]['n'], prop.property_view_id)) {
             // non-compliant dataset
             datasets[1]['data'].push(item);
+          } else {
+            // unknown dataset
+            datasets[2]['data'].push(item);
+          }
+        });
+        non_compliant = datasets.find(ds => ds.label === "non-compliant");
 
-            // add whisker annotation
-            // only when we are displaying the non-compliant metric (energy or emission)
-            // don't add whisker if data is in range for that metric or it looks bad
-            let add = false
-            metric_type = $scope.configs.chart_metric === 0 ? $scope.data.metric.energy_metric_type : $scope.data.metric.emission_metric_type;
-            if (item['x'] && item['y'] && item['target']) {
-              if ((metric_type === 1 && (item['target'] < item['y'])) || (metric_type === 2 && (item['target'] > item['y']))) {
-                add = true
-              }
-            }
+        // Rank
+        if($scope.configs.chart_xaxis === "Ranked"){
```

Comment:
above, L250, we're calculating x, then recalculating here should it be ranked. idk how to only do it once, as  the normal x value must be calculated in the for loop so we have access to the property, and the ranked must happen _after_ so it can sort. any thoughts?